### ISSUE TITLE
SNCosmo chi-squared are too large

### ISSUE THREAD

**Issue by djperrefort** [mwvgroup/Photometric-Classification#24]
The chi-squared values we are getting from SNCosmo are much too large. After some investigation with @lgalbany and @yiz143 it appears that SNCosmo is including data points outside the phase range of the model in the chi-squared calculation. PR #23 introduces (among other things) the notebook `4_chi_squared_calculation_comparison.ipynb` which demonstrates this issue. Once PR #23 is merged into master we will need to replace the chi-squared calculation with our own. The relevant portion of the code for this issue is the very end of the function `analysis_pipeline._lc_fitting._fit_funcs.fit_lc`.

Todo:
- [x] Implement our own chi-squared calculation
- [x] Delete old fit results (but not the priors) and re-run the analysis pipeline on CSP.
- [x] Generate a new classification plot (See `5_apply_classification.ipynb`)
- [ ] Manually inspect the CSP light curve fits (the 4 parameter salt2 fits are already done).
